tests/test_semantic_memory.py

```
@@ -0,0 +1,53 @@
+        self.assertIn("tone", summary)
+        self.assertIn("embedding", summary)
+        self.assertGreater(len(summary["keywords"]), 0)
+
+    def test_store_add_and_retrieve(self):
+        # Add two summaries
+        s1 = summarize("Experience with Python and ML.", {"section": "Exp1"})
+        s2 = summarize("Leadership in AI projects.", {"section": "Exp2"})
+        self.store.add(s1)
+        self.store.add(s2)
+        # Retrieve similar to a query containing "machine learning"
+        results = retrieve_relevant(["machine", "learning"], self.store, k=2)
```

Ruff (0.15.14)

[error] Undefined name `retrieve_relevant`

(F821)